packages/material-ui/src/Divider/Divider.js

```diff
@@ -1,181 +1,252 @@
+    ...(styleProps.children &&
+      styleProps.orientation === 'vertical' &&
+      styles.withChildrenVertical),
+    ...(styleProps.textAlign === 'right' &&
+      styleProps.orientation !== 'vertical' &&
+      styles.textAlignRight),
+    ...(styleProps.textAlign === 'left' &&
+      styleProps.orientation !== 'vertical' &&
+      styles.textAlignLeft),
+    [`&.${dividerClasses.wrapper}`]: styles.wrapper,
+    [`&.${dividerClasses.wrapper} .${dividerClasses.wrapperVertical}`]: styles.wrapperVertical,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const {
+    classes,
+    variant,
+    absolute,
+    flexItem,
+    light,
+    orientation,
+    children,
+    textAlign,
+  } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      variant,
+      absolute && 'absolute',
+      flexItem && 'flexItem',
+      light && 'light',
+      orientation === 'vertical' && 'vertical',
+      children && 'withChildren',
+      children && orientation === 'vertical' && 'withChildrenVertical',
+      textAlign === 'right' && orientation !== 'vertical' && 'textAlignRight',
+      textAlign === 'left' && orientation !== 'vertical' && 'textAlignLeft',
+    ],
+    wrapper: ['wrapper', orientation === 'vertical' && 'wrapperVertical'],
+  };
+
+  return composeClasses(slots, getDividerUtilityClass, classes);
+};
+
+const DividerRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiXyz',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(
+  ({ theme }) => ({
+    /* Styles applied to the root element. */
     margin: 0, // Reset browser default style.
     flexShrink: 0,
     borderWidth: 0,
     borderStyle: 'solid',
     borderColor: theme.palette.divider,
     borderBottomWidth: 'thin',
-  },
-  /* Styles applied to the root element if `absolute={true}`. */
-  absolute: {
-    position: 'absolute',
-    bottom: 0,
-    left: 0,
-    width: '100%',
-  },
-  /* Styles applied to the root element if `variant="inset"`. */
-  inset: {
-    marginLeft: 72,
-  },
-  /* Styles applied to the root element if `variant="fullWidth"`. */
-  fullWidth: {},
-  /* Styles applied to the root element if `light={true}`. */
-  light: {
-    borderColor: alpha(theme.palette.divider, 0.08),
-  },
-  /* Styles applied to the root element if `variant="middle"`. */
-  middle: {
-    marginLeft: theme.spacing(2),
-    marginRight: theme.spacing(2),
-  },
-  /* Styles applied to the root element if `orientation="vertical"`. */
-  vertical: {
-    height: '100%',
-    borderBottomWidth: 0,
-    borderRightWidth: 'thin',
-  },
-  /* Styles applied to the root element if `flexItem={true}`. */
-  flexItem: {
-    alignSelf: 'stretch',
-    height: 'auto',
-  },
-  /* Styles applied to the root element if divider have text. */
-  withChildren: {
-    display: 'flex',
-    whiteSpace: 'nowrap',
-    textAlign: 'center',
-    border: 0,
-    '&::before, &::after': {
-      position: 'relative',
+  }),
+  ({ styleProps }) => ({
```

Comment:
Please no need for callbacks if there are no pseudo or class selectors. Let's use callback only when necessary

Reply:
Apologies, still getting the hang of when to separate callbacks and when not to!